components/dashboard/src/components/QuickStart.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useEffect, useState } from "react";
+import { parseError, redirectToAuthorize, redirectToOIDC } from "../provider-utils";
+import { useHistory, useLocation } from "react-router";
+import { AppLoading } from "../app/AppLoading";
+import { Link } from "react-router-dom";
+import { authProviderClient, userClient } from "../service/public-api";
+
+const parseErrorFromSearch = (search: string): string => {
+    const searchParams = new URLSearchParams(search);
+    const message = searchParams.get("message");
+    if (message?.startsWith("error:")) {
+        const parsed = parseError(message);
+        return typeof parsed === "string" ? parsed : `${parsed.error}: ${parsed.description}`;
+    }
+
+    return "";
+};
+
+const QuickStart: FC = () => {
+    const [error, setError] = useState(parseErrorFromSearch(window.location.search));
+    const history = useHistory();
+    const { hash } = useLocation();
+
+    useEffect(() => {
+        if (error) {
+            return;
+        }
+
+        const fetchData = async () => {
+            const user = (await userClient.getAuthenticatedUser({}).catch(() => undefined))?.user;
+            const authProviderDescriptions = await authProviderClient
```

Comment:
Using hooks with queries was really inconsistent because we never want to cache here.
It would be useful to try caching these after the initial login, but there are issues while logging in with stale data (both about the user's `identities` and `authProviderDescriptions` had issues invalidating in time).

On the flipside, this adds some time to the quickstart initialization. I measured this to be negligible on my connection and about 1 second on a good 3G connection (emulated with the Chrome DevTools).